### PR TITLE
NG-5259 Add video resolution props

### PR DESCRIPTION
**Pull request by elmennani89**

### CODE REVIEW

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 1]
```diff
@@ -546,6 +558,8 @@ public void connectToRoom() {
 
         connectOptionsBuilder.enableDominantSpeaker(this.dominantSpeakerEnabled);
 
+        connectOptionsBuilder.encodingParameters(new EncodingParameters(16, this.videoDimensions.width <= 640 ? 512 : 0));
```
Comment (locomocoroco) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java]: can you add the comment from original ionic pr (i think there were a few right?)

File: ios/RCTTWVideoModule.m [hunk 2]
```diff
@@ -450,17 +460,23 @@ -(NSMutableDictionary*)convertLocalVideoTrackStats:(TVILocalVideoTrackStats *)st
...
       builder.networkQualityEnabled = true;
       builder.networkQualityConfiguration = [ [TVINetworkQualityConfiguration alloc] initWithLocalVerbosity:TVINetworkQualityVerbosityMinimal remoteVerbosity:TVINetworkQualityVerbosityMinimal];
     }
 
+    if(videoParams[@"width"] && videoParams[@"height"]){
+      NSInteger width = [videoParams[@"width"] integerValue];
+      NSInteger height = [videoParams[@"height"] integerValue];
+      kRCTTWVideoAppCameraSourceDimensions.width = width;
+      kRCTTWVideoAppCameraSourceDimensions.height = height;
+      builder.encodingParameters = [[TVIEncodingParameters alloc] initWithAudioBitrate: 16 videoBitrate:(width <= 640) ? 512 : 5000];
```
Comment (javipascual) [inline ios/RCTTWVideoModule.m]: please use param VideoQuality.LOW like ionic

Comment (javipascual) [reply ios/RCTTWVideoModule.m]: in ionic we also have something for downstream bitrate, please copy paste that too.

**Review by javipascual — CHANGES_REQUESTED** [review]

File: ios/RCTTWVideoModule.m (hunk 2 above)
Comment (javipascual) [reply ios/RCTTWVideoModule.m]: in ios is not 512

File: ios/RCTTWVideoModule.m [hunk 3]
```diff
@@ -471,10 +471,10 @@ -(NSMutableDictionary*)convertLocalVideoTrackStats:(TVILocalVideoTrackStats *)st
       builder.preferredVideoCodecs = @[ [TVIH264Codec new] ];
     }
 
-    if(encodingParameters[@"audioBitrate"] || encodingParameters[@"videoBitrate"]){
-      NSInteger audioBitrate = [encodingParameters[@"audioBitrate"] integerValue];
```
Comment (locomocoroco) [inline ios/RCTTWVideoModule.m]: so we could already pass audio and video bitrate?

**Review by locomocoroco — APPROVED** [review]

File: ios/RCTTWVideoModule.m (hunk 3 above)
Comment (javipascual) [reply ios/RCTTWVideoModule.m]: if we can pass it from RN, then it is better, sure

**Review by javipascual — CHANGES_REQUESTED** [review]

File: ios/RCTTWVideoModule.m (hunk 3 above)
Comment (elmennani89) [reply ios/RCTTWVideoModule.m]: bit rates can be passed only in iOS, this is why I chose common interface for both

File: ios/RCTTWVideoModule.m (hunk 2 above)
Comment (elmennani89) [reply ios/RCTTWVideoModule.m]: in capacitor plugin we also used 512: https://github.com/media4care/capacitor-twilio-video/blob/master/android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java (hunk 1 above)
Comment (elmennani89) [reply android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java]: added

File: ios/RCTTWVideoModule.m (hunk 3 above)
Comment (javipascual) [reply ios/RCTTWVideoModule.m]: rather to delete code that is working and tested by many people I would add these params to android, then we don't need to pass this quality one, no?

Comment (elmennani89) [reply ios/RCTTWVideoModule.m]: done

File: android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java [hunk 4]
```diff
@@ -546,6 +568,10 @@ public void connectToRoom() {
 
         connectOptionsBuilder.enableDominantSpeaker(this.dominantSpeakerEnabled);
 
+        // limit upstream bitrate depending on VideoQuality param
+        // 16 kbps for audio, 512 kbps for low quality video, 0 for max
```
Comment (javipascual) [inline android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java:572]: this comment is not needed anymore

**Review by javipascual — APPROVED** [review]

**Review by locomocoroco — APPROVED** [review]

**Pull request merged**